persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -38,12 +41,30 @@ private[lagom] class SlickProvider(
 
   val autoCreateTables: Boolean = createTables.getBoolean("auto")
 
+  private val asyncExecConfig =
+    new AsyncExecutorConfig(system.settings.config.getConfig("jdbc-defaults.slick.async-executor"))
+
   if (dbApi != null) {
     // Work around https://github.com/playframework/playframework/issues/7262
     // Set the system property
     System.setProperty(Context.PROVIDER_URL, "/")
-    // Rebind the datasource
-    new InitialContext().rebind("DefaultDS", dbApi.database("default").dataSource)
+
+    // the data source as configured by Play
+    val dataSource = dbApi.database("default").dataSource
+
+    // the slick db configured with an async executor
+    val slickDb =
+      SlickDbProvider(
+        dataSource,
+        system.settings.config.getConfig("jdbc-defaults.slick.async-executor")
```

Comment:
I don't think it should be reading from the `jdbc-defaults` namespace. That's meant only to populate default values for the specific configurations for read journal, write journal, snapshot store and read-side. This implementation no longer allows them to be different.

We also need to think about what customizations might already exist in the wild. If someone has already overridden `jndiName` then we need to fall back to the previous behavior and should not create an extra database here.

Reply:
I will move the async-executor settings to db.default. That's the same place where we have `hikaricp` and `bonecp` settings.

Not sure either if this is the best place, though.

Reply:
Regarding customizations, I don't think we were so flexible. 

Previous code was:

```scala
if (dbApi != null) {
    // Work around https://github.com/playframework/playframework/issues/7262
    // Set the system property
    System.setProperty(Context.PROVIDER_URL, "/")
    // Rebind the datasource
    new InitialContext().rebind("DefaultDS", dbApi.database("default").dataSource)
  }
```

And the reason why we needed to rebound was the play playframework/playframework/issues/7262 issue. If someone had configured their data source to be on another JNDI name, then chances are big that they have found it problematic because from time to time they would get an error from that. 

I would live this for another ticket (occasion). We should be able to configure different connection/thread pool per plugin component + userland read-side code, but I consider this as out-of-scope for this ticket.

Reply:
I'm changing my mind about it. 
It's not so hard to provide this functionally. 

I will send a commit for it.

Reply:
That code was added in Lagom 1.4 to work around a bug in the Play 2.6 milestones... I'm glad we found it before releasing Lagom 1.4, because you're right that it would have broken customizations. I think it was always intended to be removed before release but we could have easily missed it.